transports/http/endpoints/api/contacts/endpoints.go

```diff
@@ -0,0 +1,199 @@
+package contacts
+
+import (
+	"net/http"
+
+	"github.com/bitcoin-sv/spv-wallet-web-backend/domain"
+	"github.com/bitcoin-sv/spv-wallet-web-backend/domain/contacts"
+	"github.com/bitcoin-sv/spv-wallet/models"
+
+	"github.com/rs/zerolog"
+
+	"github.com/bitcoin-sv/spv-wallet-web-backend/transports/http/auth"
+	router "github.com/bitcoin-sv/spv-wallet-web-backend/transports/http/endpoints/routes"
+
+	"github.com/gin-gonic/gin"
+)
+
+type handler struct {
+	cService contacts.ContactsService
+	log      *zerolog.Logger
+}
+
+// NewHandler creates new endpoint handler.
+func NewHandler(s *domain.Services, log *zerolog.Logger) router.ApiEndpoints {
+	return &handler{
+		cService: *s.ContactsService,
+		log:      log,
+	}
+}
+
+// RegisterApiEndpoints registers routes that are part of service API.
+func (h *handler) RegisterApiEndpoints(router *gin.RouterGroup) {
+	user := router.Group("/contact")
+
+	user.PUT("/:paymail", h.upsertContact)
+	user.PATCH("/accepted/:paymail", h.acceptContact)
+	user.PATCH("/rejected/:paymail", h.rejectContact)
+	user.PATCH("/confirmed", h.confirmContact)
+	user.GET("/search", h.getContacts)
+	user.POST("/totp", h.generateTotp)
+}
+
+// Get all user contacts.
+//
+//	@Summary Get all contacts.
+//	@Tags contact
+//	@Produce json
+//	@Success 200 {object} []models.Contact
+//	@Router /api/v1/contacts/search [get]
+//	@Param data body SearchContact true "Conditions for filtering contacts"
+func (h *handler) getContacts(c *gin.Context) {
+	var req SearchContact
+	err := c.Bind(&req)
+	if err != nil {
+		h.log.Error().Msgf("Invalid payload: %s", err)
+		c.JSON(http.StatusBadRequest, "Invalid request. Please check conditions and metadata")
+		return
+	}
+
+	// Get user transactions.
+	txs, err := h.cService.GetContacts(c.Request.Context(), c.GetString(auth.SessionAccessKey), req.Conditions, &req.Metadata, nil)
+	if err != nil {
+		h.log.Error().Msgf("An error occurred while trying to get a list of contacts: %s", err)
+		c.JSON(http.StatusInternalServerError, "An error occurred while trying to get a list of contacts")
+		return
+	}
+
+	c.JSON(http.StatusOK, txs)
+}
+
+// Upsert contact.
+//
+//	@Summary Create or update a contact.
+//	@Tags contact
+//	@Produce json
+//	@Success 200
+//	@Router /api/v1/contact/{paymail} [put]
+//	@Param data body UpsertContact true "Upsert contact data"
+func (h *handler) upsertContact(c *gin.Context) {
+	paymail := c.Param("paymail")
+
+	var req UpsertContact
+	err := c.Bind(&req)
+	if err != nil {
+		h.log.Error().Msgf("Invalid payload: %s", err)
+		c.JSON(http.StatusBadRequest, "Invalid request. Please check contact details")
+		return
+	}
+
+	_, err = h.cService.UpsertContact(c.Request.Context(), c.GetString(auth.SessionAccessKey), paymail, req.FullName, &req.Metadata)
+	if err != nil {
```

Comment:
Differentiating errors from the server is probably overkill for the reference implementation, but if we don't do it, we should return `InternalServerError`, in my opinion